### ISSUE TITLE
require-atomic-updates reports false positive when the assigned variable is used in a lambda and function call

### ISSUE THREAD

**Issue by Liu233w** [eslint/eslint#11959]
**Tell us about your environment**

I've built a simple example in [ESLint Demo](https://eslint.org/demo/#eyJ0ZXh0IjoiYXN5bmMgZnVuY3Rpb24gbWFpbigpIHtcclxuICBsZXQgYSA9IHNvbWVGdW5jKClcclxuICBhKCgpPT5hKSAvLyBjaGFuZ2UgaXQgdG8gYGYoKCk9PmEpYCwgYGEoYSlgIG9yIGBhKCgpPT5iKWAgd2lsbCBhbGwgcGFzcyB0aGUgbGludFxyXG4gIGEgPSBhd2FpdCBmdW5jKCkgLy8gUG9zc2libGUgcmFjZSBjb25kaXRpb246IGBhYCBtaWdodCBiZSByZWFzc2lnbmVkIGJhc2VkIG9uIGFuIG91dGRhdGVkIHZhbHVlIG9mIGBhYC5cclxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6OSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJyZXF1aXJlLWF0b21pYy11cGRhdGVzIjoyfSwiZW52Ijp7fX19), my configuration is included in it.

* **ESLint Version:**
6.0.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 9,
        "sourceType": "script",
        "ecmaFeatures": {}
    },
    "rules": {
        "require-atomic-updates": 2
    },
    "env": {}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
async function main() {
  let a = someFunc()
  a(()=>a) // change it to `f(()=>a)`, `a(a)` or `a(()=>b)` will all pass the lint
  a = await func() // Possible race condition: `a` might be reassigned based on an outdated value of `a`.
}
```

**What did you expect to happen?**
Pass without error

**What actually happened? Please include the actual, raw output from ESLint.**
Possible race condition: `a` might be reassigned based on an outdated value of `a`.

**Comment by mysticatea** [comment]
Thank you for your report.

In fact, as based on the discussion in #11774, this is intentional behavior. ESLint cannot know when `()=>a`  function will be called, so the rule handles it as a possible race condition.

**Comment by Liu233w** [comment]
Thank you for your reply, @mysticatea .

According to the discussion, statements like `f(()=>a)` should also be marked as a possible race condition. But in the demo above, only `a(()=>a)` is reported, while `f(()=>a)` does not. Is that an intended behavior?
